pkg/rbac/sharedData.go

```diff
@@ -51,6 +55,68 @@ var managedClusterResourceGvr = schema.GroupVersionResource{
 	Resource: "managedclusters",
 }
 
+func (shared *SharedData) getPropertyTypes(cache *Cache, ctx context.Context) (map[string]string, error) {
+
+	// original query: select distinct key, jsonb_typeof(value) as datatype FROM search.resources,jsonb_each(data);
+	var selectDs *goqu.SelectDataset
+
+	//define schema:
+	schemaTable := goqu.S("search").Table("resources")
+
+	//data expression to get value and key
+	jsb := goqu.L("jsonb_each(?)", goqu.C("data"))
+
+	//select from these datasets
+	ds := goqu.From(schemaTable, jsb)
+
+	//select statement with orderby and distinct clause
+	selectDs = ds.Select(goqu.L("key"), goqu.L("jsonb_typeof(?)",
+		goqu.C("value")).As("datatype")).Distinct()
+
+	query, params, sqlerr := selectDs.ToSQL()
+	if sqlerr != nil {
+		klog.Errorf("Error building Search query: %s", sqlerr.Error())
+	}
+
+	klog.V(5).Infof("Query for property datatypes: [%s] ", query)
+	rows, err := cache.pool.Query(ctx, query, params...)
+	if err != nil {
+		klog.Errorf("Error resolving query [%s] with args [%+v]. Error: [%+v]", query, err)
+		return nil, err
+	}
+	defer rows.Close()
+	resourceTypeMap := make(map[string]string)
```

Comment:
Move this to the top and return that instead of nil to avoid nil pointer error later.